easybuild/framework/easyblock.py

```diff
@@ -3303,6 +3303,19 @@ def regex_for_lib(lib):
 
         return fail_msg
 
+    def sanity_check_mod_files(self):
+        """
+        Check installation for Fortran .mod files
+        """
+        self.log.debug(f"Checking for .mod files in install directory {self.installdir}...")
+        mod_files = glob.glob(os.path.join(self.installdir, '**', '*.mod'), recursive=True)
+
+        fail_msg = None
+        if mod_files:
+            fail_msg = ".mod files (%s) found in the installation." % ', '.join(mod_files)
```

Comment:
@branfosj Can we try to filter out false positives here?

I was thinking about checking the output of the `file` command, if that's available.
Not sure if that makes sense...

Reply:
I'm not sure what we would want to exclude by using `file`.

```
$ file 2.3.0-GCCcore-11.2.0/lib64/kim-api/mod/kim_unit_system_module.mod
2.3.0-GCCcore-11.2.0/lib64/kim-api/mod/kim_unit_system_module.mod:                     gzip compressed data, from Unix, original size 57651

$ file Clang/15.0.5-GCCcore-11.3.0/include/flang/omp_lib.mod
Clang/15.0.5-GCCcore-11.3.0/include/flang/omp_lib.mod:                       UTF-8 Unicode (with BOM) text

$ file json-fortran/8.3.0-GCCcore-12.2.0/include/json_value_module.mod
json-fortran/8.3.0-GCCcore-12.2.0/include/json_value_module.mod:     gzip compressed data, from Unix, original size 201261

$ file CPLEX/20.10-GCCcore-10.2.0/opl/examples/opl/cplex_staticLex/staticlex3.mod
CPLEX/20.10-GCCcore-10.2.0/opl/examples/opl/cplex_staticLex/staticlex3.mod: ASCII text
```

Reply:
As discussed, we can refine this later to try and weed out false positives without having to whitelist...